File: iniciante/python/1164-numero-perfeito.py
def numero_perfeito():
    numero_de_casos = int(input())
    for i in range(numero_de_casos):
        numero = int(input())
        eh_perfeito(numero)


def eh_perfeito(n):
    divisores = list()
    for j in range(1, int(n/2+1)):
        if n % j == 0:
            divisores.append(j)
    soma_divisores = sum(divisores)
    if soma_divisores == n:
        print(f'{n} eh perfeito')
    else:
        print(f'{n} nao eh perfeito')


numero_perfeito()

File: iniciante/c/1164-numero-perfeito.c
#include <stdio.h>
#include <stdbool.h>

void eh_perfeito(int n){
    int i, soma = 0;
    for(i = 1; i < n / 2 + 1; i++){
        if (n % i == 0){
            soma += i;
        }
    }
    if (soma == n){
        printf("%d eh perfeito\n", n);
    }
    else{
        printf("%d nao eh perfeito\n", n);
    }
}

int main(){
    int n_teste, i, numero;
    scanf("%d", &n_teste);
    for (i = 0; i < n_teste; i++){
        scanf("%d", &numero);
        eh_perfeito(numero);
    }
    return 0;
}
